### ISSUE TITLE
TableCPD type checking too strict

### ISSUE THREAD

**Issue by finnhacks42** [pgmpy/pgmpy#716]
TableCPD init method requires variable_card to be an int. This makes it incompatible with model.get_cardinality(.), which returns a numpy.int64

The fix is trivial, replace

`if not isinstance(variable_card, int):`

with 
`import numbers`
`if not isinstance(variable_card,numbers.Integral)`

should I create a pull request?

**Comment by yashu-seth** [comment]
@finnhacks42 model.get_cardinality calls self.get_cpds which looks for the cardinality attribute in the CPD object. I don't understand why it produces a numpy.int64 type. 
It would be great if you can share a reproducible example. It would make things clearer.

**Comment by finnhacks42** [comment]
The following prints class 'numpy.int64' and then throws a TypeError 

```
model = BayesianModel([('X','Y')])
cpd_x = TabularCPD(variable='X', variable_card=2, values=[[0.6, 0.4]])
cpd_y = TabularCPD(variable='Y', variable_card=2,
                   values=[[0.95, 0.2],
                           [0.05, 0.8]],
                   evidence=['X'],
                   evidence_card=[2])
model.add_cpds(cpd_x,cpd_y)
model.check_model()

x_card = model.get_cardinality('X')

print(type(x_card))
cpd_new = TabularCPD(variable='X', variable_card=x_card, values=[[0.6, 0.4]])
```

In general should we be doing instance checks explicitly against native python int? A numpy int works just as well. It would seem more pythonic to accept either.

**Comment by khalibartan** [comment]
@finnhacks42 Yes, indeed we should be allowing all type integers, native python and numpy integer types. You can send a pull request for that.

**Comment by ankurankan** [comment]
Fixed in #717
